builder/googlecompute/step_create_instance.go

```diff
@@ -40,19 +40,24 @@ func (c *Config) createInstanceMetadata(sourceImage *Image, sshPublicKey string)
 		instanceMetadata[sshMetaKey] = sshKeys
 	}
 
-	// Wrap any startup script with our own startup script.
+	startupScript := instanceMetadata[StartupScriptKey]
 	if c.StartupScriptFile != "" {
 		var content []byte
 		content, err = ioutil.ReadFile(c.StartupScriptFile)
 		if err != nil {
 			return nil, err
 		}
-		instanceMetadata[StartupWrappedScriptKey] = string(content)
-	} else if wrappedStartupScript, exists := instanceMetadata[StartupScriptKey]; exists {
-		instanceMetadata[StartupWrappedScriptKey] = wrappedStartupScript
+		startupScript = string(content)
+	}
+	instanceMetadata[StartupScriptKey] = startupScript
+
+	// Wrap any startup script with our own startup script.
+	if startupScript != "" && c.WrapStartupScriptFile.True() {
+		instanceMetadata[StartupWrappedScriptKey] = startupScript
```

Comment:
nitpick but I don't think we need to re-set this since we're already setting in line 52

Reply:
Good catch!

Reply:
I think this is then only thing left before we can merge. Looks good.

Reply:
So after review it is setting the wrapped startup script. I updated the order a little so that it is clearer what is being set.